internal/dcache/rpc/server/handler.go

```diff
@@ -47,43 +57,894 @@ var _ service.ChunkService = &ChunkServiceHandler{}
+}
+
+// TODO:: integration: sample method, will be later removed after integrating with cluster manager
+// call cluster manager to create the rvID map from config
+func getRvIDMap() map[string]*rvInfo {
+	return make(map[string]*rvInfo)
+}
+
+// TODO:: integration: sample method, will be later removed after integrating with cluster manager
+// call cluster manager to get chunk size from config
+func getChunkSize() int64 {
+	return 4 * 1024 * 1024 // 4MB
+}
+
+// TODO:: integration: sample method, will be later removed after integrating with cluster manager
+// call cluster manager to get mvs-per-rv from config
+func getMVsPerRV() int64 {
+	return 10
+}
+
+// TODO:: integration: sample method, will be later removed after integrating with cluster manager
+// call cluster manager helper method to get available disk space for the given path
+func getAvailableDiskSpace(path string) (int64, error) {
+	return 0, nil
+}
+
+// TODO:: integration: sample method, will be later removed after integrating with cluster manager
+// call cluster manager helper method to get the node ID of this node
+func getNodeUUID() string {
+	return "node-uuid" // TODO: get the node uuid from the config
+}
+
+// check the if the chunk address is valid
+// - check if the rvID is valid
+// - check if the cache dir exists
+// - check if the MV is valid
+func (h *ChunkServiceHandler) checkValidChunkAddress(address *models.Address) error {
+	if address == nil || address.FileID == "" || address.RvID == "" || address.MvName == "" {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: Invalid chunk address %+v", *address)
+		return rpc.NewResponseError(rpc.InvalidRequest, fmt.Sprintf("invalid chunk address %+v", *address))
+	}
+
+	// check if the rvID is valid
+	rvInfo, ok := h.rvIDMap[address.RvID]
+	if !ok || rvInfo == nil {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: Invalid rvID %s", address.RvID)
+		return rpc.NewResponseError(rpc.InvalidRVID, fmt.Sprintf("invalid rvID %s", address.RvID))
+	}
+
+	cacheDir := rvInfo.cacheDir
+	if cacheDir == "" || !common.DirectoryExists(cacheDir) {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: Cache dir not found for RV %s", rvInfo.rvName)
+		return rpc.NewResponseError(rpc.InternalServerError, fmt.Sprintf("cache dir not found for RV %s", rvInfo.rvName))
+	}
+
+	// check if the MV is valid
+	mvPath := filepath.Join(cacheDir, address.MvName)
+	if err := rvInfo.isMvValid(mvPath); err != nil {
+		log.Err("ChunkServiceHandler::checkValidChunkAddress: MV %s is not hosted by RV %s [%s]", address.MvName, rvInfo.rvName, err.Error())
+		return rpc.NewResponseError(rpc.MVNotHostedByRV, fmt.Sprintf("MV %s is not hosted by RV %s [%s]", address.MvName, rvInfo.rvName, err.Error()))
+	}
+
+	return nil
+}
+
+// get the RVInfo from the RV name
+func (h *ChunkServiceHandler) getRVInfoFromRvName(rvName string) *rvInfo {
+	var rvInfo *rvInfo
+	for _, info := range h.rvIDMap {
+		if info == nil {
+			continue
+		}
```

Comment:
we should assert for it, instead of handling it

Reply:
Done, updated